### DISCUSSION TITLE
simple Luckysheet component

### DISCUSSION THREAD

**Discussion by retsyo** [zauberzeug/nicegui#2387] [Ideas / Feature Requests]
[Luckysheet](https://github.com/dream-num/Luckysheet)

> is an online spreadsheet like excel that is powerful, simple to configure, and completely open source.

[CrystalWindSnake](https://github.com/CrystalWindSnake), not me, supplies a simple wrap to [Luckysheet](https://github.com/dream-num/Luckysheet), which can be found under `nicegui-examples/tree/master/src/luckysheet示例` [here](https://gitee.com/carson_add/nicegui-examples). You can clone the code and run the `main.py`.  

However, too much has not been bound in `luckysheet.py` yet.

![luckysheet](https://github.com/zauberzeug/nicegui/assets/7960913/9abfe3a5-a60f-41a6-b5e5-871f2044ef82)

**Comment by retsyo** [comment]
BTW
1. is there any written doc on how to bind JavaScript element for NiceGUI? Yes, there is `nicegui-main\examples\custom_vue_component` but it could be far from a tutorial for users who do not know JavaScript well
2. could it be possible to coin a tool to auto-wrap JavaScript component to NiceGUI? A similiar one could be [SWIG](https://www.swig.org/), which supplies Python, Ruby, and more languages' binding according to C/C++ code

Thanks

**Reply by falkoschindler** [reply]
Yes, the integration of third-party libraries is an ongoing topic (see #1454) and I wish to find a simpler way to wrap existing libraries.

And no, the documentation on creating custom components is pretty sparse (or non existing) at the moment. 😕

**Comment by falkoschindler** [comment]
Wow, that looks impressive! Maybe such a powerful spreadsheet component for NiceGUI like [Luckysheet](https://github.com/dream-num/Luckysheet) (which has been renamed to [Univer](https://github.com/dream-num/univer)) justifies an own repository with a separate development team, because the feature set might get quite overwhelming. It might be similar to [nicegui-highcharts](https://github.com/zauberzeug/nicegui-highcharts), which we separated just recently for licensing reasons.
